packages/kit/src/DropdownButton.tsx

```diff
@@ -1,70 +1,59 @@
-import React from 'react'
-import { Button, Sheet, YStack } from 'tamagui'
-import { z } from 'zod'
+/** @jsxImportSource @final-ui/react */
 
-const DropdownButtonProps = z.object({
-  id: z.string().optional(),
-  onSelect: z.function().args(z.string()).optional(),
-  buttonProps: z.any().optional(),
-  options: z
-    .array(
-      z.object({
-        key: z.string(),
-        label: z.string(),
-        icon: z.any().optional(),
-      })
-    )
-    .optional(),
-})
+import { extend, Only, setStateAction, state, WithServerProps } from '@final-ui/react'
+import {
+  Button,
+  Sheet,
+  SheetFrame,
+  SheetHandle,
+  SheetOverlay,
+  YStack,
+} from '@final-ui/tamagui/server'
 
-export function DropdownButton(props: z.infer<typeof DropdownButtonProps>) {
-  const { options } = props
-  const [open, setOpen] = React.useState(false)
+type Props = {
+  onSelect?: (item: string) => void
+  button?: JSX.Element
+  options?: {
+    key: Only<string>
+    label: string
+  }[]
+}
+
+export function DropdownButton(props: WithServerProps<Props>) {
```

Comment:
Previously, we would write:
```
export const RiseSwitchField = createComponentDefinition<typeof SwitchField>('RiseSwitchField')
```
to create a server component definition that can be used with JSX on the server.

Now, that we write the server component from scratch, we need a way to decorate its incoming props too. This is needed so that users can pass e.g. "StateDataState" where type is "string".

Example:

Consider the DropdownButton:
```
<DropdownButton onSelect={() => {}} />
```

in this case, `onSelect` is a function. However, you should be also able to write:
```
<DropdownButton onSelect={event(() => {}, { timeout: 1000 })} />
```

Same for e.g. `label`:
```
<DropdownButton options={[{ label: 'option-a' }]} />
```

can be `state`:
```
const label = state('my-label')
<DropdownButton options={[{ label }]} />
```

In practice, this just adds more type safety, but there's nothing required from user side. If you pass e.g. `label` prop straight to Tamagui (imported from `@final-ui/tamagui/server`), its type is already updated to accept `string | StateDataState` so there is no issue.